### PR TITLE
Duplicate v1alpha1 AuthenticationConfiguration to v1beta1

### PR DESCRIPTION
**Pull request by aramase**
#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:

Adds apiserver.config.k8s.io/v1beta1 AuthenticationConfiguration types

part of https://github.com/kubernetes/enhancements/issues/3331

#### Does this PR introduce a user-facing change?

```release-note
kube-apiserver: the AuthenticationConfiguration type accepted in `--authentication-config` files has been promoted to `apiserver.config.k8s.io/v1beta1`.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3331-structured-authentication-configuration
```

/sig auth
/triage accepted
/priority important-soon

### CODE REVIEW

**Comment by enj** [comment]
Opened #123719 to promote the feature gate.

**Comment by enj** [comment]
@aramase probably good to make the same change as #123721 for the beta struct.

File: staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1beta1/types.go
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1beta1/types.go:1]: `diff --width=250 --side-by-side staging/src/k8s.io/apiserver/pkg/apis/apiserver/{v1alpha1,v1beta1}/types.go` shows these are identical to the v1alpha1 types

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1beta1/types.go:1]: verified there were no v1alpha1 manual defaults or conversions that needed copying over

**Review by liggitt — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**